### PR TITLE
Event Type: Title not displayed in the mobile view

### LINKED ISSUE #7414: Event Type: Title not displayed in the mobile view
Untitled (1)

Should be
Untitled

https://www.figma.com/file/xk4HOxtSI82J0F7enMxeak/Cal---Live?node-id=118%3A95390&t=8zS8cVe85u9bcLZj-4

### PR DESCRIPTION
**Pull request by gitstart-calcom**
## What does this PR do?

Fixes #7414 

## Demo
https://www.loom.com/share/7da38f262af242c58691a348e80dbbfb

### CODE REVIEW

**Review by CarinaWolli — COMMENTED** [review]
I am not sure if we even need to hide the heading anywhere. Availability also misses the heading ([Figma Designs Availability](https://www.figma.com/file/xk4HOxtSI82J0F7enMxeak/Cal---Live?node-id=26%3A46706&t=M5JCUPFWHtYk1XNX-0)) 

@sean-brydon or @Jaibles do you know if we need to hide it in some places?

**Comment by gitstart-calcom** [comment]
Hi @CarinaWolli, the heading was breaking on some of the pages that's why we thought maybe it was not supposed to show everywhere.
Screenshot 2023-02-27 at 1 35 01 PM

Screenshot 2023-02-27 at 1 33 03 PM

But if it's to be shown on all pages, we will make the changes accordingly

**Comment by gitstart-calcom** [comment]
Also, how should we handle editable headings, like on the `Availability` page?

**Comment by CarinaWolli** [comment]
> Hi @CarinaWolli, the heading was breaking on some of the pages that's why we thought maybe it was not supposed to show everywhere.

Okay. Then it would probably be best to solve that and show the headings everywhere


> Also, how should we handle editable headings, like on the Availability page?

Tagging @Jaibles

**Comment by gitstart-calcom** [comment]
@Jaibles We have now too many buttons in the Availability page heading
Screenshot 2023-03-03 at 8 54 53 AM

**Comment by Jaibles** [comment]
> @Jaibles We have now too many buttons in the Availability page heading
> 
> Screenshot 2023-03-03 at 8 54 53 AM

See my image above and check out a single event type for how to handle it.

**Comment by gitstart-calcom** [comment]
> > @Jaibles We have now too many buttons in the Availability page heading
> > Screenshot 2023-03-03 at 8 54 53 AM
> 
> See my image above and check out a single event type for how to handle it.

Do you mean to move the buttons to a dropdown?

**Comment by CarinaWolli** [comment]
> Do you mean to move the buttons to a dropdown?

Exactly, same as we have it for event types just with "delete" and "set to default"

Screenshot 2023-03-03 at 11 46 44

**Comment by gitstart-calcom** [comment]
Thank you @Jaibles and @CarinaWolli for the clarifications. We are working on the changes

**Comment by emrysal** [comment]
Converted to draft whilst changes are in progress @gitstart-calcom

**Comment by gitstart-calcom** [comment]
Hi @Jaibles, noticed this weird behavior of the toggle in the detail page of the `Routing Form`
https://www.loom.com/share/c608415d33624978821a91d43e45e375

Do we really need that toggle on the detail page?
If yes, we are moving the toggle to dropdown for the mobile view but there's no label for the toggle. What should it be labeled as?
Screenshot 2023-03-09 at 2 01 22 PM

File: apps/web/components/eventtype/EventTypeSingleLayout.tsx [hunk 1]
```diff
@@ -195,6 +195,7 @@ function EventTypeSingleLayout({
       backPath="/event-types"
       title={eventType.title + " | " + t("event_type")}
       heading={eventType.title}
+      showHeading
```
Comment (hariombalhara) [inline apps/web/components/eventtype/EventTypeSingleLayout.tsx:198]: I am confused why is there a showHeading prop? I would always want to show the heading.

Edit: https://github.com/calcom/cal.com/issues/5420 I found out that this was requirement to hide the heading in this case which is why hiding in mobile was implemented.

File: packages/features/shell/Shell.tsx [hunk 2]
```diff
@@ -805,8 +806,9 @@ export function ShellMain(props: LayoutProps) {
               {props.heading && (
                 <h3
                   className={classNames(
-                    "font-cal max-w-28 sm:max-w-72 md:max-w-80 hidden truncate text-xl font-semibold tracking-wide text-black md:block xl:max-w-full",
-                    props.smallHeading ? "text-base" : "text-xl"
+                    "font-cal max-w-28 sm:max-w-72 md:max-w-80 hidden truncate text-lg font-semibold tracking-wide text-black sm:text-xl md:block xl:max-w-full",
```
Comment (hariombalhara) [inline packages/features/shell/Shell.tsx:809]:
```suggestion
                    "font-cal max-w-28 sm:max-w-72 md:max-w-80 truncate text-lg font-semibold tracking-wide text-black sm:text-xl md:block xl:max-w-full",
```

**Review by hariombalhara — CHANGES_REQUESTED** [review]
Left some comments

**Comment by Jaibles** [comment]
When I test this on live, the toggle is taking a really long time to respond, if it even responds at all.
https://user-images.githubusercontent.com/4536123/226644587-47e8baa3-d81c-475e-8134-f03b2e23fd2c.mp4

File: apps/web/components/apps/App.tsx [hunk 3]
```diff
@@ -369,7 +369,7 @@ export default function App(props: {
   dependencies?: string[];
 }) {
   return (
-    <Shell smallHeading isPublic heading={<ShellHeading />} backPath="/apps" withoutSeo>
+    <Shell smallHeading isPublic showHeadingAlways heading={<ShellHeading />} backPath="/apps" withoutSeo>
```
Comment (PeerRich) [inline apps/web/components/apps/App.tsx:372]: i dont think this prop is a good solution

**Comment by gitstart-calcom** [comment]
Hi @sean-brydon, we made the requested changes, please take a look 🙏

**Comment by CarinaWolli** [comment]
>  When I test this on live, the toggle is taking a really long time to respond, if it even responds at all.
> https://user-images.githubusercontent.com/4536123/226644587-47e8baa3-d81c-475e-8134-f03b2e23fd2c.mp4

That's actually already like that now, @hariombalhara do you know why the toggle is not working (or slow)?

And @sean-brydon could you take a look again at the new changes? I think UI wise looks good

**Comment by sean-brydon** [comment]
> > When I test this on live, the toggle is taking a really long time to respond, if it even responds at all.
> > https://user-images.githubusercontent.com/4536123/226644587-47e8baa3-d81c-475e-8134-f03b2e23fd2c.mp4
> 
> That's actually already like that now, @hariombalhara do you know why the toggle is not working (or slow)?
> 
> And @sean-brydon could you take a look again at the new changes? I think UI wise looks good

Much better 🙌  Looks good from my POV :)

**Review by CarinaWolli — APPROVED** [review]
LGTM! @hariombalhara can you review and check if your requested changes are fixed? And @sean-brydon can you give a review for @calcom/ui?

**Comment by hariombalhara** [comment]
> @hariombalhara can you explain what this toggle actually does so I can come up with a reasonable line of copy to put in as a tooltip and for the label in the tooltip?

Not sure how I missed it. That toggle is supposed to make a Routing Form toggle the availability of Routing Form to public.
You can work on a Routing Form peacefully after toggling it off and then toggle it back on.

**Comment by hariombalhara** [comment]
cc @Jaibles  ^^

File: packages/ui/components/form/switch/Switch.tsx [hunk 4]
```diff
@@ -29,40 +30,76 @@ const Switch = (
...
   return (
     <Wrapper tooltip={props.tooltip}>
-      <div
-        className={cx(
-          "flex h-auto w-auto flex-row items-center",
-          fitToHeight && "h-fit",
-          classNames?.container
-        )}>
-        <PrimitiveSwitch.Root
+      {props.isRoutingForm ? (
```
Comment (hariombalhara) [inline packages/ui/components/form/switch/Switch.tsx:33]: I don't know why we need `isRoutingForm` here. It is a generic Switch component which shouldn't know what a Routing FOrm is

**Review by hariombalhara — DISMISSED** [review]
Left a blocking comment. 

Reviewing more in the meantime.

**Comment by hariombalhara** [comment]
@CarinaWolli @Jaibles Fixed the toggle not working issue. It was actually trpc query invalidation issue.

File: packages/app-store/routing-forms/components/FormActions.tsx [hunk 5]
```diff
@@ -410,15 +410,24 @@ export const FormAction = forwardRef(function FormAction<T extends typeof Button
...
         if (!routingForm) {
           return <></>;
         }
         return (
-          <div {...restProps} className="hover:bg-emphasis self-center rounded-md p-2">
+          <div
+            {...restProps}
+            className={classNames(
+              "sm:hover:bg-subtle self-center p-2",
+              label === "Disable Form"
```
Comment (hariombalhara) [inline packages/app-store/routing-forms/components/FormActions.tsx:427]: This isn't the correct way. It's too fragile.

**Comment by Jaibles** [comment]
Much better 👏

**Review by zomars — APPROVED** [review]
Code LGTM!

**Pull request merged**